### PR TITLE
[16.0][MIG] base_edifact: Migration to 16.0

### PR DESCRIPTION
**Pull request by QuocDuong1306**
Start migrating from:
- https://github.com/OCA/edi/pull/779

### CODE REVIEW

File: base_edifact/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    # "preloadable": True,
+    "external_dependencies": {
+        "python": ["pydifact"],
+        "bin": [],
+    },
+    "depends": [
+        "edi_party_data_oca",
```
Comment (simahawk) [inline base_edifact/__manifest__.py]: hmm I don't think we want edi framework to be used here

File: base_edifact/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    # "preloadable": True,
+    "external_dependencies": {
+        "python": ["pydifact"],
+        "bin": [],
+    },
+    "depends": [
+        # "edi_party_data_oca"
```
Comment (simahawk) [inline base_edifact/__manifest__.py]: this change should not belong to this commit

File: base_edifact/tests/test_base_edifact.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    def setUpClass(cls):
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+
+    def setUp(self):
+        super().setUp()
+        self.base_edifact_model = self.env["base.edifact"]
```
Comment (simahawk) [inline base_edifact/tests/test_base_edifact.py]: this should go to setUpClass

File: base_edifact/tests/test_base_edifact.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        return fd.read()
+
+
+class TestBaseEdifact(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
```
Comment (simahawk) [inline base_edifact/tests/test_base_edifact.py]: useless overrides

**Review by simahawk — CHANGES_REQUESTED** [review]
Would be nice to see some more test cov

**Comment by QuocDuong1306** [comment]
Thank @simahawk , I updated

File: base_edifact/models/edifact.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
+import datetime
```
Comment (simahawk) [inline base_edifact/models/edifact.py:5]: (c) missing

**Review by simahawk — APPROVED** [review]
LG overall

File: base_edifact/models/edifact.py (hunk 5 above)
Comment (simahawk) [reply base_edifact/models/edifact.py:5]: @QuocDuong1306 can you fix this pls?  Then we are ready to go.

Comment (QuocDuong1306) [reply base_edifact/models/edifact.py:5]: Done here, thank @simahawk
